crates/cxx-qt-lib-headers/include/qt_types.h

```diff
@@ -31,12 +31,30 @@ struct rust::IsRelocatable<QColor> : std::true_type
 };
 static_assert(QTypeInfo<QColor>::isRelocatable);
 
+template<>
+struct rust::IsRelocatable<QString> : std::true_type
+{
+};
+static_assert(QTypeInfo<QString>::isRelocatable);
+
 // Ensure that types have the alignment and size we are expecting
 static_assert(alignof(QColor) <= alignof(std::size_t[2]),
               "unexpectedly large QColor alignment");
 static_assert(sizeof(QColor) == sizeof(std::size_t[2]),
               "unexpected QColor size");
 
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
+static_assert(alignof(QString) <= alignof(std::size_t[2]),
+              "unexpectedly large QString alignment");
+static_assert(sizeof(QString) == sizeof(std::size_t[2]),
+              "unexpected QString size");
```

Comment:
This is incorrect:
```
warning: In file included from /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxxbridge/sources/cxx-qt-lib/src/types/qcolor.rs.cc:1:
warning: /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxx-qt-lib/include/qt_types.h:62:31: error: static assertion failed: unexpected QString size
warning:    62 | static_assert(sizeof(QString) == sizeof(std::size_t[2]),
warning:       |               ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
warning: In file included from /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxxbridge/sources/cxx-qt-lib/src/types/qdate.rs.cc:1:
warning: /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxx-qt-lib/include/qt_types.h:62:31: error: static assertion failed: unexpected QString size
warning:    62 | static_assert(sizeof(QString) == sizeof(std::size_t[2]),
warning:       |               ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
warning: In file included from /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxxbridge/sources/cxx-qt-lib/src/types/qdatetime.rs.cc:1:
warning: /home/runner/cxx-qt/build/cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-5d4bdfe03d4a5e41/out/cxx-qt-lib/include/qt_types.h:62:31: error: static assertion failed: unexpected QString size
warning:    62 | static_assert(sizeof(QString) == sizeof(std::size_t[2]),
warning:       |               ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
```

How did you determine the sizes of these Qt types?

Reply:
I haven't yet for Qt 6, it was a blind guess :-D

Reply:
Fixed, Qt 5 has one pointer as a member, Qt 6 has one member which contains two pointers and a size_t